spring-cloud-gcp-starters/spring-cloud-gcp-starter-sql/src/main/java/org/springframework/cloud/gcp/sql/SqlCredentialFactory.java

```diff
@@ -0,0 +1,69 @@
+/*
+ *  Copyright 2017 original author or authors.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package org.springframework.cloud.gcp.sql;
+
+import java.io.FileInputStream;
+import java.io.IOException;
+
+import com.google.api.client.auth.oauth2.Credential;
+import com.google.api.client.googleapis.auth.oauth2.GoogleCredential;
+import com.google.cloud.sql.CredentialFactory;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+
+import org.springframework.cloud.gcp.core.autoconfig.GcpContextAutoConfiguration;
+
+/**
+ * Returns the credentials that are written to a system property by the Cloud SQL starter.
+ *
+ * <p>
+ * Since the sockets factory creates an instance of this class by reflection and without any
+ * arguments, the credential location must be in a place that this class knows without any context.
+ *
+ * @author João André Martins
+ */
+public class SqlCredentialFactory implements CredentialFactory {
+
+	public static final String CREDENTIAL_LOCATION_PROPERTY_NAME =
+			"GOOGLE_CLOUD_SQL_CREDS_LOCATION";
+
+	private static final Log LOGGER = LogFactory.getLog(SqlCredentialFactory.class);
+
+	@Override
+	public Credential create() {
+		// TODO(joaomartins): Consider supporting Spring Resources as credential locations. There
+		// would need to be a way to create or inject a Spring context here statically, to load
+		// the resource from there.
+		String credentialResourceLocation = System.getProperty(CREDENTIAL_LOCATION_PROPERTY_NAME);
```

Comment:
If you worry about Spring Context, you can just set this system property from some bean which would create a temporary file based on the `Resource`.
Otherwise that is always bad idea to store Spring context in the static scope. You shouldn't forget that your JVM may manage several applications, which indeed may use the same classes.